protocol-jackson/src/main/java/io/camunda/zeebe/protocol/jackson/record/ValueTypes.java

```diff
@@ -0,0 +1,142 @@
+import io.camunda.zeebe.protocol.jackson.record.MessageStartEventSubscriptionRecordValueBuilder.ImmutableMessageStartEventSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.MessageSubscriptionRecordValueBuilder.ImmutableMessageSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.ProcessBuilder.ImmutableProcess;
+import io.camunda.zeebe.protocol.jackson.record.ProcessEventRecordValueBuilder.ImmutableProcessEventRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.ProcessInstanceCreationRecordValueBuilder.ImmutableProcessInstanceCreationRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.ProcessInstanceRecordValueBuilder.ImmutableProcessInstanceRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.ProcessInstanceResultRecordValueBuilder.ImmutableProcessInstanceResultRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.ProcessMessageSubscriptionRecordValueBuilder.ImmutableProcessMessageSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.TimerRecordValueBuilder.ImmutableTimerRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.VariableDocumentRecordValueBuilder.ImmutableVariableDocumentRecordValue;
+import io.camunda.zeebe.protocol.jackson.record.VariableRecordValueBuilder.ImmutableVariableRecordValue;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.DeploymentDistributionIntent;
+import io.camunda.zeebe.protocol.record.intent.DeploymentIntent;
+import io.camunda.zeebe.protocol.record.intent.ErrorIntent;
+import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
+import io.camunda.zeebe.protocol.record.intent.JobBatchIntent;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.intent.MessageIntent;
+import io.camunda.zeebe.protocol.record.intent.MessageStartEventSubscriptionIntent;
+import io.camunda.zeebe.protocol.record.intent.MessageSubscriptionIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessEventIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceCreationIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceResultIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessMessageSubscriptionIntent;
+import io.camunda.zeebe.protocol.record.intent.TimerIntent;
+import io.camunda.zeebe.protocol.record.intent.VariableDocumentIntent;
+import io.camunda.zeebe.protocol.record.intent.VariableIntent;
+import java.util.EnumMap;
+import java.util.Map;
+
+/**
+ * Provides a mapping of all {@link ValueType} to their concrete implementations. Can be extended in
+ * the future to also provide a way for converting base {@link
+ * io.camunda.zeebe.protocol.record.RecordValue} types recursively to immutable objects.
```

Comment:
This is more of a TODO for myself, I guess. I can remove it.

To clarify: type info can be extended to include the expected RecordValue interface (e.g. `VariableRecordValue.class`) and the mapping to the immutable equivalent (e.g. `ImmutableVariableRecordValue.class` via `new VariableRecordValueBuilder().from().build()`). This will give you a type safe way to easily map values to their immutable counter parts. You can then use this to easily recursively copy, say, a mutable `Record<?>`, converting it to an `ImmutableRecord<?>` _and_ also its `getValue()` property to the correct immutable value. But this isn't part of this PR, and while it makes it a bit tedious for the user to do the copy manually, I think that's OK for now.

Reply:
Thanks for explaining. I'd rather have it left out of the javadoc in that case. Feel free to turn it into a `//todo(@npepinpe):...` comment though.